schema/m3-room.schema.json

```diff
@@ -0,0 +1,1985 @@
+                  "type": "number",
+                  "title": "Minimum Speed in Tiles",
+                  "description": "The minimum amount of effective runway tiles in other room needed for this strat."
+                }
+              }
+            },
+            "comeInWithBombBoost": {
+              "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInWithBombBoost",
+              "type": "object",
+              "title": "Come In With Bomb Boost",
+              "description": "Represents that Samus must come through this door with a horizontal bomb boost.",
+              "required": [],
+              "additionalProperties": false,
+              "properties": {}
+            },
+            "comeInWithDoorStuckSetup": {
+              "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInWithDoorStuckSetup",
+              "type": "object",
+              "title": "Come In With Door Stuck Setup",
+              "description": "Represents that Samus must come through this door in a way that allows getting stuck in the door (e.g. to set up an X-ray climb).",
+              "required": [],
+              "additionalProperties": false,
+              "properties": {}
+            },
+            "comeInSpeedballing": {
+              "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInSpeedballing",
+              "type": "object",
+              "title": "Come In Speedballing",
+              "description": "Represents that Samus must enter the room either in a speedball from the previous room, or in a process of running, jumping, or falling into a speedball.",
+              "required": ["runway"],
+              "additionalProperties": false,
+              "properties": {
+                "runway": {
+                  "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInSpeedballing/properties/runway",
+                  "$ref": "#/definitions/runway",
+                  "description": "Runway in the current room that is available to complete the speedball."
+                }
+              }
+            },
+            "comeInWithRMode": {
+              "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInWithRMode",
+              "type": "object",
+              "title": "Come in With R-Mode",
+              "description": "Represents that Samus must obtain R-mode coming through this door.",
+              "required": [],
+              "additionalProperties": false,
+              "properties": {}
+            },
+            "comeInWithGMode": {
+              "$id": "#/definitions/strat/properties/entranceCondition/properties/comeInWithGMode",
+              "type": "object",
+              "title": "Come in With G-Mode",
+              "description": "Represents that Samus must have or obtain G-mode coming through this door.",
+              "required": [
+                "mode",
+                "morphed"
```

Comment:
I don't think `morphed` should be required unless a majority of GMode strats use `morphed`.  And that may be the case.

Reply:
I think it's fine either way. If we change it to be not required, again maybe it can be done in a separate PR to keep the history cleaner?